pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -106,8 +106,6 @@ def __init__(
         else:  # without MPI
             self._qubit_state = self._state_dtype()(self.num_wires)
 
-        self.reset_state()
-
```

Comment:
We remove reset here in favour of adding the zero-state init everywhere in the C++ API. I'm a bit concerned about this change tbh :thinking: Could you share benchmark results?

Reply:
In fact, `self.reset_state()` calls the `BaseType::initSV();` from the python layer, while now we call `BaseType::initSV()` directly in the C++ layer;.

Reply:
I support the answer of @multiphaseCFD :+1:

Reply:
@LuisAlfredoNu I assumed you didn't get my point! We don't need to run `initSV` in the constructor of the state-vector class. We should be able to avoid setting all the elements to zero. Nevertheless the performance boost up of this change is minimal while the `reset_state` method provides users to reset the state-vector from the Python API.

As discussed offline with @multiphaseCFD, we should investigate the removal of `initSV` calls in the constructors to be able to merge this PR.